### PR TITLE
Odroid xu4 : first port

### PR DESCRIPTION
**Pull request by nadenislamarre**
Please make sure your PR is ready to be merged !
- [ ] You added the changes in CHANGELOG.md
- [x] You choose the right repository branch to make the PR
- [ ] You described the PR as below

You can now create the odroid xu4 sdcard in 3 commands from scratch ;-)
still kodi, n64, dreamcast, ... not working. I'll work on that later.

make recalbox-odroidxu4_defconfig
make
sudo dd if=output/images/recalbox/recalbox.sdcard of=/dev/mmcblk0 bs=40M; sync

Signed-off-by: Nicolas Adenis-Lamarre nicolas.adenis.lamarre@gmail.com

### CODE REVIEW

**Comment by substring** [comment]
nice job @nadenislamarre  ! I'm really impresed :)

It raises a few questions though :
- what's the difference between xu3 and xu4 ? why make a xu4 only port if both architectures are "the same", knowing that you had to take some parts from xu3 ?
- what's left to do ? Adapt packages to XU4 ?
- didn't you mention a package that can make an img without root privileges ?

And many more ... I'll let @digitalLumberjack be the most critic :p

**Comment by nadenislamarre** [comment]
The xu3 and xu4 are softwarely compatible.
The xu4 is smaller, more powerfull and cheaper.
The xu3 is not more to sell, it has been replaced by xu4.

The image works for xu3 and xu4.
But you can read that i use the kernel made for xu3 for example.
hardkernel provides a 4.2 kernel for xu4, but it still doesn't include the mali drivers, so, i stay on the xu3 kernel for the moment.

Left to do :
- fix the linux patches (to add some joysticks fixes)
- diff config_xu4 and config_rpi3 to add removed packages (compilation issues, and gpu issues : n64, reicast, ...)
- make kodi to work
- make the power on/off button to work
- find a replacement for omxplayer which is only for the rpi for the boot animation
- see if we have things to change in our packages (recalbox .mk files) to make more optimisations or make nicer configuration (use BR_CPU_HAS_NEON to add --enable-neon for exemple instead of if A8 or 15 do --enable-neon) and such things.

currently, it is fully functionnal.
It's just limited by some emu and kodi missing.

**Comment by nadenislamarre** [comment]
@substring :
have a look to the following commands :

make recalbox-odroidxu4_defconfig
make
sudo dd if=output/images/recalbox/recalbox.sdcard of=/dev/mmcblk0 bs=40M; sync

output/images/recalbox/recalbox.sdcard is generated without the root account.
It is generated instead of boot.tar.xz and root.tar.xz.
In fact, i ask to buildroot to build a rootfs.ext4
then, i use genimage to build the .sdcard.
Note that the xu4 can use both sdcard or emmc, but my build is for sdcard only.
I could add an option in buildroot to build for sd or emmc. The difference is only 1 bloc as far as i remember when writting in the card the blobs for the secure boot - a very small change, but that require a emmc to test.

Root is just used in the commands to write on the sdcard.

**Comment by nadenislamarre** [comment]
reicast support added.

**Comment by substring** [comment]
The makefile patch is wrong : there shouldn't be any reference to /home/subs/blablabla. Usually such patchs are set by the .mk with a sed. Anyway i have to simplify the .mk and remove unecessary compile flags

**Comment by nadenislamarre** [comment]
i agree
but that is the same wronginess as patch 001makefile
/home/subs doenst come from my computer ;-)
however it works.

**Comment by substring** [comment]
Oh my bad, it was just supposed to add the path to libzip headers With a /opt/vc path that should be sed-ed by the .mk. Thatnks for pointing me this out

**Comment by nadenislamarre** [comment]
poweroff button handled.

i've created a project from evtest to handle that simpyly :
https://github.com/nadenislamarre/evwait

File: package/recalbox-system/Config.in [hunk 1]
```diff
@@ -2,3 +2,25 @@ config BR2_PACKAGE_RECALBOX_SYSTEM
     bool "Recalbox System"
     help
       Install the recalbox system files
+
+choice
```
Comment (digitalLumberjack) [inline package/recalbox-system/Config.in:6]: This should be directly in the def-config. No choices for the end user as one could enable xu4 here, but compile in armv6.

**Pull request merged**